cmd/runtimecfg/node-ip.go

```diff
@@ -252,7 +259,10 @@ func getSuitableIPs(retry bool, vips []net.IP, preferIPv6 bool, networkType stri
 		}
 
 		log.Errorf("Failed to find a suitable node IP")
-		time.Sleep(time.Second)
+		time.Sleep(time.Second * time.Duration(timerLoop))
+		if timerLoop == maxSecondsToSuitableIPsLoop {
```

Comment:
This part is fragile, let me show you why - we start from `1` and then we add `addSecondsToSuitableIPsLoop` every time we loop. So in our case we start with `1`, add `2` so after the 1st iteration we have `3`. Then we again add `2` so we have `5`... And so on, till we reach `maxSecondsToSuitableIPsLoop` which now is `300`.

But this assumes that this repetitive addition will get you to exactly `maxSecondsToSuitableIPsLoop` at some point. So what happens if someone in the future changes `addSecondsToSuitableIPsLoop` from `2` to let's say `7`. This way you will never have `timerLoop == maxSecondsToSuitableIPsLoop` returning `true`, because of how `1+7+7+7+7...` works

So tldr it would be safer to have `timerLoop >  maxSecondsToSuitableIPsLoop`

Reply:
that make sense.